### PR TITLE
Beta release 3.0.0-beta1

### PR DESCRIPTION
**Pull request by cerebrl**
Notes:

- Updated CHANGELOG.md
- Updated test timers (20 to 60 seconds) as there were some occasional failures due to timeout
- Updated dependencies
- For those that use "dark mode", I added a "dark mode" for the e2e test pages for eye comfort
- Cleaned up some dependencies (removed unneeded package and script that required it)
- Added build and e2e tests back to `git push` hook

### CODE REVIEW

File: tests/e2e/app/authn-basic/index.html [hunk 1]
```diff
@@ -8,6 +8,25 @@
...
+      }
+      a:visited {
+        color: lavender;
+      }
+      a:hover {
+        color: lightskyblue;
+      }
+    }
+  </style>
+
```
Comment (cerebrl) [inline tests/e2e/app/authn-basic/index.html:29]: Just adding dark mode to the E2E tests for those that have dark mode set for the OS. The pure white page will not longer blind those that have dark mode enabled :)

File: package.json [hunk 2]
```diff
@@ -34,7 +33,7 @@
   "husky": {
     "hooks": {
       "pre-commit": "tsc && npm run lint",
-      "pre-push": "npm run test:unit && npm run test:integration"
+      "pre-push": "npm run test:unit && npm run test:integration && npm run build && npm run test:e2e"
```
Comment (spetrov) [inline package.json:36]: 👍

File: package.json [hunk 3]
```diff
@@ -16,7 +16,6 @@
     "clean": "shx rm -rf ./bundles ./docs ./lib ./lib-esm ./coverage",
     "clean:all": "npm run clean && shx rm -rf ./node_modules ./package-lock.json && git clean -fX -e \"!*.pem\"",
     "docs": "typedoc",
-    "docs:watch": "watch \"npm run docs\" ./src",
```
Comment (cerebrl) [inline package.json:19]: Removed it as it was never used and required a package that had a vulnerability in it.

File: package.json [hunk 4]
```diff
@@ -79,7 +78,6 @@
     "typescript": "^4.0.0",
     "util": "^0.12.3",
     "uuid": "^8.3.0",
-    "watch": "^1.0.2",
```
Comment (cerebrl) [inline package.json:82]: Removed due to vulnerability.

File: tests/e2e/utilities/setup-and-go.ts [hunk 5]
```diff
@@ -46,7 +46,7 @@ export async function setupAndGo(
       permissions: config && config.allowGeo ? ['geolocation'] : [],
     });
     const page = await context.newPage();
-    page.setDefaultTimeout(10 * 1000); // 10 seconds (make be less than the Jest config timeout)
+    page.setDefaultTimeout(50 * 1000); // 50 seconds (make be less than the Jest config timeout)
```
Comment (cerebrl) [inline tests/e2e/utilities/setup-and-go.ts:49]: Will help eliminate purely timeout issues with e2e tests.

File: tests/jest.e2e.config.js [hunk 6]
```diff
@@ -18,7 +18,18 @@ module.exports = {
...
+      'jest-junit',
+      {
+        outputDirectory: `reports/${Date.now()}`,
+        suiteNameTemplate: '{filename}',
+        classNameTemplate: '{filename}',
+        includeConsoleOutput: true,
+      },
+    ],
+  ],
+  testTimeout: 60 * 1000, // 60 seconds (must be more than the setup-and-go.ts timeout)
```
Comment (cerebrl) [inline tests/jest.e2e.config.js:33]: Fixes timeout issue and just reformats the object here. No functional changes.

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**